File: 01_StartStop/StartStop_CSharp/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Windows.Forms;
using System.Runtime.InteropServices;

using SolidEdgeFramework;

namespace Cust01
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }
        
        SolidEdgeFramework.Application oApp;
        Type typ;

        private void cmdStart_Click(object sender, EventArgs e)
        {
            try
            {
                //typ = Type.GetTypeFromProgID("SolidEdge.Application");
                //oApp =  (SolidEdgeFramework.Application)Activator.CreateInstance(typ);
                oApp = (SolidEdgeFramework.Application)Marshal.GetActiveObject("SolidEdge.Application");
            }
		    catch (Exception ex)
            {
			    MessageBox.Show(ex.Message);
            }
        }

        private void cmdShow_Click(object sender, EventArgs e)
        {
            oApp.Visible = true;
        }

        private void cmdHide_Click(object sender, EventArgs e)
        {
            oApp.Visible = false;
        }

        private void cmdVersion_Click(object sender, EventArgs e)
        {
            label1.Text = oApp.Name + " - " + oApp.Version;
        }

        private void cmdAppend_Click(object sender, EventArgs e)
        {
            oApp.StatusBar += txtStatus.Text;
        }

        private void chkStatus_CheckedChanged(object sender, EventArgs e)
        {
            oApp.StatusBarVisible = chkStatus.Checked;
        }

        private void chkEdge_CheckedChanged(object sender, EventArgs e)
        {
            oApp.EdgeBarVisible = chkEdge.Checked;
        }

        private void cmdQuit_Click(object sender, EventArgs e)
        {
            oApp.Quit();
        }

        private void txtStatus_TextChanged(object sender, EventArgs e)
        {

        }

        private void label1_Click(object sender, EventArgs e)
        {

        }
    }
}

File: 01_StartStop/StartStop_VB.Net/Form1.vb
Imports System.Runtime.InteropServices
Imports SolidEdgeFramework

Public Class Form1
	Dim oApp As SolidEdgeFramework.Application
	Dim typ As Type

	Private Sub cmdStart_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles cmdStart.Click
		Try
            oApp = Marshal.GetActiveObject("SolidEdge.Application")
            'oApp = CreateObject("SolidEdge.Application")
		Catch ex As Exception
			MessageBox.Show(ex.Message)
		End Try
	End Sub

	Private Sub cmdShow_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles cmdShow.Click
		oApp.Visible = True
	End Sub

	Private Sub cmdHide_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles cmdHide.Click
		oApp.Visible = False
	End Sub

	Private Sub cmdVersion_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles cmdVersion.Click
		label1.Text = oApp.Name & " - " & oApp.Version
	End Sub

	Private Sub cmdAppend_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles cmdAppend.Click
		oApp.StatusBar += txtStatus.Text
	End Sub

	Private Sub chkStatus_CheckedChanged(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles chkStatus.CheckedChanged
		oApp.StatusBarVisible = chkStatus.Checked
	End Sub

	Private Sub chkEdge_CheckedChanged(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles chkEdge.CheckedChanged
		oApp.EdgeBarVisible = chkEdge.Checked
	End Sub

	Private Sub cmdQuit_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles cmdQuit.Click
		oApp.Quit()
	End Sub
End Class
